### ISSUE TITLE
[BUG] Deployment yaml gets mangled at hostAlias after form edition

### ISSUE THREAD

**Issue by baptisterajaut** [rancher/rancher#45329]
Hey, i noticed i had this annoying bug.

**Rancher Server Setup**
- Rancher version: 2.8.3
- Installation option : Docker Install

**User Information**
- What is the role of the user logged in? : Admin

**Describe the bug**
When creating a host alias, everything is well.
Example , here's the yaml of a service which works : 
```yaml
     hostAliases:
        - hostnames:
            - mysql
          ip: 192.168.10.10
      restartPolicy: Always
```

However when i went and edited it with the form, i can't apply it because 
`Deployment in version "v1" cannot be handled as a Deployment: json: cannot unmarshal string into Go struct field HostAlias.spec.template.spec.hostAliases.hostnames of type []string`

And as it stands, indeed the hostAliases spec is flipped: 

```yaml
      hostAliases:
        - ip: 192.168.10.10
          hostnames: mysql
#        - hostnames:
#            - string
#          ip: string
      hostNetwork: false
      hostname: my-db-hostname
      restartPolicy: Always
```

**Result**

Have to re-set the yaml by hand

**Expected Result**
It shouldn't be necessary

**Screenshots**

**Additional context**
I added the network hostname. I did not add the hostNetwork property.

**Comment by rak-phillip** [comment]
@baptisterajaut thanks for raising this issue and helping to make Rancher better!

I attempted to reproduce the issue you've described, but I was unable to generate the same error. 

To help us better understand and triage this issue, could you provide a minimal manifest that generates the error? Can you also provide steps on how you're editing the host alias through the form?

**Comment by baptisterajaut** [comment]
I can replicate this issue 100%.

- Create a deployment with a host alias to the pod

```yaml
spec:
      affinity: {}
      containers:
        - image: nginx:latest
          imagePullPolicy: Always
          name: container-0
          resources: {}
          securityContext:
            allowPrivilegeEscalation: false
            privileged: false
            readOnlyRootFilesystem: false
            runAsNonRoot: false
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
      dnsConfig: {}
      dnsPolicy: ClusterFirst
      hostAliases:
        - hostnames:
            - somewhere
          ip: 1.1.1.1
```
This will deploy fine.

- Go edit that deployment, and add a hostname to the pod

![image](https://github.com/rancher/rancher/assets/814492/2a03a60c-8e2e-46de-bb8e-93229a093fde)

- Press save, and enjoy the `Deployment in version "v1" cannot be handled as a Deployment: json: cannot unmarshal string into Go struct field HostAlias.spec.template.spec.hostAliases.hostnames of type []string`

```yaml
 spec:
      containers:
        - image: nginx:latest
          imagePullPolicy: Always
          name: container-0
          securityContext:
            allowPrivilegeEscalation: false
            privileged: false
            readOnlyRootFilesystem: false
            runAsNonRoot: false
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
          _init: false
          __active: true
          resources: {}
      dnsPolicy: ClusterFirst
      hostAliases:
        - ip: 1.1.1.1
          hostnames: somewhere
#        - hostnames:
#            - string
#          ip: string
      hostNetwork: false
      hostname: myalias
      restartPolicy: Always
      schedulerName: default-scheduler
```

- Basically, the "somewhere" is just missing a line return and a dash before.

**Comment by rak-phillip** [comment]
@baptisterajaut thanks for the additional information, the issue has been confirmed. 

This appears to be an issue with workload resources (deployments, replicasets, jobs, etc...) and doesn't appear on the pods form.

@gaktive transferring this issue to the Dashboard repo for additional triage.
